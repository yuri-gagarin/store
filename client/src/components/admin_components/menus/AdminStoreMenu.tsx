import React, { useState, useEffect } from "react";
import { Menu, MenuItemProps } from "semantic-ui-react";
// routing //
import { withRouter, RouteComponentProps } from "react-router-dom"
// css imports //
import "./css/adminStoreMenu.css";

interface Props extends RouteComponentProps {

}
const AdminStoreMenu: React.FC<Props> = ({ history }): JSX.Element => {
  const [ activeItem, setActiveItem ] = useState<string>("view_all");
  const [ menuOpen, setMenuOpen ] = useState<boolean>(false);

  const handleItemClick = (e: React.MouseEvent, { name }: MenuItemProps): void => {
    setActiveItem(String(name));
    const baseUrl: string = "/admin/home/my_store";
    switch (name) {
      case "view_all": 
        history.push(baseUrl + "/all");
        break;
      case "create":
        history.push(baseUrl + "/create");
        break;
      case "manage":
        history.push(baseUrl + "/manage");
        break;
      default: history.push(baseUrl);
    }
  }
  useEffect(() => {
    setTimeout(() => {
      setMenuOpen(true);
    }, 200);
  }, []);

  return (
    <div>
      <Menu tabular className={ menuOpen ? "adminStoreMenu storeMenuOpen" : "adminStoreMenu" }>
      <Menu.Item
          name='view_all'
          content="View All"
          active={activeItem === 'view_all'}
          onClick={handleItemClick}
        />
        <Menu.Item
          name='create'
          content="Create Store"
          active={activeItem === 'create'}
          onClick={handleItemClick}
        />
        <Menu.Item
          name='manage'
          content="Manage"
          active={activeItem === 'manage'}
          onClick={handleItemClick}
        />
      </Menu>
    </div>
  )
};

export default withRouter(AdminStoreMenu);