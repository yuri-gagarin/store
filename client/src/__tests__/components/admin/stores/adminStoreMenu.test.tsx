import React from "react";
import { mount, ReactWrapper } from "enzyme";
import AdminStoreMenu from "../../../../components/admin_components/menus/AdminStoreMenu";
import { MemoryRouter } from "react-router-dom";
import { Menu } from "semantic-ui-react";
describe("AdminStoreMenu component render tests", () => {
  let wrapper: ReactWrapper;
  beforeAll(() => {
    wrapper = mount(
      <MemoryRouter>
        <AdminStoreMenu dispatch={jest.fn}/>
      </MemoryRouter>
    )
  })
  describe("Default component render", () => {
    it("Should properly render and match snapshot", () => {
      expect(wrapper.find(AdminStoreMenu).render()).toMatchSnapshot();
    });
    it("Should have 3 'Menu.Item' components", () => {
      const menuItems = wrapper.find(Menu.Item);
      expect(menuItems.length).toEqual(3);
    });
    it("Should correctly respond to 'View All' 'Menu.Item' click", () => {
      window.scrollTo = jest.fn;
      const menuItems = wrapper.find(Menu.Item)
      menuItems.at(0).simulate("click");
      wrapper.update();
      // assert correct rendering //
      expect(wrapper.find(Menu.Item).at(0).props().active).toEqual(true);
      expect(wrapper.find(Menu.Item).at(1).props().active).toEqual(false);
      expect(wrapper.find(Menu.Item).at(2).props().active).toEqual(false);
    });
    it("Should correctly respond to 'Creat' 'Menu.Item' click", () => {
      window.scrollTo = jest.fn;
      const menuItems = wrapper.find(Menu.Item)
      menuItems.at(1).simulate("click");
      wrapper.update();
      // assert correct rendering //
      expect(wrapper.find(Menu.Item).at(0).props().active).toEqual(false);
      expect(wrapper.find(Menu.Item).at(1).props().active).toEqual(true);
      expect(wrapper.find(Menu.Item).at(2).props().active).toEqual(false);
    })
    it("Should correctly respond to 'Manage' 'Menu.Item' click", () => {
      window.scrollTo = jest.fn;
      const menuItems = wrapper.find(Menu.Item)
      menuItems.at(2).simulate("click");
      wrapper.update();
      // assert correct rendering //
      expect(wrapper.find(Menu.Item).at(0).props().active).toEqual(false);
      expect(wrapper.find(Menu.Item).at(1).props().active).toEqual(false);
      expect(wrapper.find(Menu.Item).at(2).props().active).toEqual(true);
      console.log(wrapper.props())
    })
  })
})