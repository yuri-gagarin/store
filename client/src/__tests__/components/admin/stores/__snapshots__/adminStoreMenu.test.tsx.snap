// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminStoreMenu component render tests Default component render Should properly render and match snapshot 1`] = `
<div
  class="adminStoreMenuFixed"
>
  <div
    class="ui tabular adminStoreMenu menu"
  >
    <a
      class="active item adminMenuLink"
    >
      View All
    </a>
    <a
      class="item adminMenuLink"
    >
      Create Store
    </a>
    <a
      class="item adminMenuLink"
    >
      Manage
    </a>
  </div>
</div>
`;
